uvicorn/protocols/websockets/websockets.py

```diff
@@ -24,9 +24,12 @@ def set_header(key, val):
         return
 
     # Retrieve any subprotocols to be negotiated with the consumer later
-    subprotocols = request_headers.get(b"sec-websocket-protocol", None)
-    if subprotocols:
-        subprotocols = subprotocols.split(b",")
+    subprotocols = []
+    for header_key, header_value in http.headers:
+        if header_key != b"sec-websocket-protocol":
+            continue
+        for subprotocol in header_value.split(b","):
+            subprotocols.append(subprotocol.decode("ascii").strip(" "))
```

Comment:
I guess just `.strip()` might make sense here. (Shouldn't be other whitespace characters, but we may as well drop 'em if there are, right?)

Reply:
Sure thing!